test/metabase/middleware_test.clj

```diff
@@ -220,7 +220,7 @@
               eventually (apply str (async/<!! (async/into [] connection)))]
           [first-second second-second (string/trim eventually)])))))
 
-
+(comment
```

Comment:
This makes me a bit nervous as it happened very infrequently before, and it sounds like now it's pretty frequent. If we can convince ourselves this is a test only problem (the tests do look a little troublesome anyway) and this is a feature we care about, we should write up an issue for fixing it, note it here and work on it soon.

Reply:
have we figured out what to do with this yet?

Reply:
Not really. Just eliminated one option. Here's @senior's take after we went through:

"One of the things I was worried about with the async processing going on with xrays and the streaming test failures we were seeing is overloading the clojure threadpool that is used whenever you invoke a future in clojure. I was thinking that pool was bounded, but I was confused, it's the agent pool that is bounded, not the pool that futures use.

After chatting some with Simon, I don't think this is the cause of our issue, but something to keep in mind is where our background/async tasks are running and where the back pressure is. In the case of sending async emails or the xray code, it's running with the unbounded clojure "soloExecutor" threadpool."